datasets/polyglot_ner/polyglot_ner.py

```diff
@@ -0,0 +1,189 @@
+         title = {{Polyglot-NER}: Massive Multilingual Named Entity Recognition},
+         journal = {{Proceedings of the 2015 {SIAM} International Conference on Data Mining, Vancouver, British Columbia, Canada, April 30- May 2, 2015}},
+         month     = {April},
+         year      = {2015},
+         publisher = {SIAM},
+}
+"""
+
+_LANGUAGES = [
+    "ca",
+    "de",
+    "es",
+    "fi",
+    "hi",
+    "id",
+    "ko",
+    "ms",
+    "pl",
+    "ru",
+    "sr",
+    "tl",
+    "vi",
+    "ar",
+    "cs",
+    "el",
+    "et",
+    "fr",
+    "hr",
+    "it",
+    "lt",
+    "nl",
+    "pt",
+    "sk",
+    "sv",
+    "tr",
+    "zh",
+    "bg",
+    "da",
+    "en",
+    "fa",
+    "he",
+    "hu",
+    "ja",
+    "lv",
+    "no",
+    "ro",
+    "sl",
+    "th",
+    "uk",
+]
+
+_LANG_FILEPATHS = {
+    lang: os.path.join(
+        "acl_datasets",
+        lang,
+        "data" if lang != "zh" else "",  # they're all lang/data/lang_wiki.conll except "zh"
+        f"{lang}_wiki.conll",
+    )
+    for lang in _LANGUAGES
+}
+
+_DESCRIPTION = """\
+Polyglot-NER
+A training dataset automatically generated from Wikipedia and Freebase the task
+of named entity recognition. The dataset contains the basic Wikipedia based
+training data for 40 languages we have (with coreference resolution) for the task of
+named entity recognition. The details of the procedure of generating them is outlined in
+Section 3 of the paper (https://arxiv.org/abs/1410.3791). Each config contains the data
+corresponding to a different language. For example, "es" includes only spanish examples.
+"""
+
+_DATA_URL = "http://cs.stonybrook.edu/~polyglot/ner2/emnlp_datasets.tgz"
+_HOMEPAGE_URL = "https://sites.google.com/site/rmyeid/projects/polylgot-ner"
+_VERSION = "1.0.0"
+
+
+class PolyglotNERConfig(datasets.BuilderConfig):
+    def __init__(self, *args, languages=None, **kwargs):
+        super().__init__(*args, version=datasets.Version("1.0.0", ""), **kwargs)
+        self.languages = languages
+
+    @property
+    def filepaths(self):
+        return [_LANG_FILEPATHS[lang] for lang in self.languages]
+
+
+class PolyglotNER(datasets.GeneratorBasedBuilder):
+    """The Polyglot-NER Dataset"""
+
+    BUILDER_CONFIGS = [
+        PolyglotNERConfig(name=lang, languages=[lang], description=f"Polyglot-NER examples in {lang}.")
+        for lang in _LANGUAGES
+    ] + [
+        PolyglotNERConfig(
+            name="combined", languages=_LANGUAGES, description=f"Complete Polyglot-NER dataset with all languages."
+        )
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "lang": datasets.Value("string"),
+                    "tokens": datasets.Sequence(datasets.Value("string")),
+                    "labels": datasets.Sequence(datasets.Value("string")),
```

Comment:
We usually use `words` and `ner` instead of `tokens` and `labels` (see LinCE, xtreme/pan-x, conll-200X)
Could you use those names too ?

Reply:
Done (I was going off W-NUT, which seems to be the one counterexample)

Reply:
I'll rename its fields as well I think